scripts/wait_beaker_dataset_model_upload_then_evaluate_model.py

```diff
@@ -0,0 +1,52 @@
+import sys
+import time
+from dataclasses import dataclass
+
+from open_instruct.utils import (
+    ArgumentParserPlus,
+    BeakerRuntimeConfig,
+    beaker_experiment_succeeded,
+    get_beaker_dataset_ids,
+    submit_beaker_eval_jobs,
+)
+
+"""
+example usage
+python scripts/wait_beaker_dataset_model_upload_then_evaluate_model.py --beaker_workload_id 01J6ASGRDTH9G9DWF9FPDWM6C1 --model_name "eval_beaker_experiment_until_finished"
+"""
+
+
+@dataclass
+class Args:
+    model_name: str
+    max_wait_time_for_beaker_dataset_upload_seconds: int = 60 * 30  # 30 minutes
+    check_interval_seconds: int = 60
+
+
+def main(args: Args, beaker_runtime_config: BeakerRuntimeConfig):
+    print(args)
+    beaker_dataset_ids = get_beaker_dataset_ids(beaker_runtime_config.beaker_workload_id)
+    print(beaker_experiment_succeeded(beaker_runtime_config.beaker_workload_id))
+
+    start_time = time.time()
+    while time.time() - start_time < args.max_wait_time_for_beaker_dataset_upload_seconds:
+        if beaker_experiment_succeeded(beaker_runtime_config.beaker_workload_id):
+            print("Experiment succeeded")
+            # NOTE: we are assuming the first beaker dataset has the model
+            # I have checked a couple of beaker jobs and found the first dataset is the model
+            # but we should check this assumption
```

Comment:
@yizhongw @jacob-morrison for multi-node job I assumed the first beaker dataset will have the model. Is this always the case?

Reply:
I checked my experiment list. It seems so. I didn't see an exception.

Reply:
my one concern is if this is the same when preemption happens, but thats sort of an edge case so happy to merge and then patch later.